{
  "Software_Repo_MappingId": null,
  "Software_RepoId": "f6b863cd-8f1c-4583-8bc5-ee59e86e5cf1",
  "Software_Name": "Python 3.12.5 (64-bit)",
  "CVEID": "CVE-2022-42919",
  "Summary": "Python 3.9.x before 3.9.16 and 3.10.x before 3.10.9 on Linux allows local privilege escalation in a non-default configuration. The Python multiprocessing library, when used with the forkserver start method on Linux, allows pickles to be deserialized from any user in the same machine local network namespace, which in many system configurations means any user on the same machine. Pickles can execute arbitrary code. Thus, this allows for local user privilege escalation to the user that any forkserver process is running as. Setting multiprocessing.util.abstract_sockets_supported to False is a workaround. The forkserver start method for multiprocessing is not the default start method. This issue is Linux specific because only Linux supports abstract namespace sockets. CPython before 3.9 does not make use of Linux abstract namespace sockets by default. Support for users manually specifying an abstract namespace socket was added as a bugfix in 3.7.8 and 3.8.3, but users would need to make specific uncommon API calls in order to do that in CPython before 3.9.",
  "LastUpdate": 0,
  "CVEE": {
    "CVEEId": "CVE-2022-42919",
    "CVSS_V2_baseScore": null,
    "CVSS_V2_Data": null,
    "CVSS_V3_baseScore": 7.8,
    "CVSS_V3_Data": "{\"version\":\"3.1\",\"baseScore\":7.8,\"accessVector\":null,\"vectorString\":\"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H\",\"authentication\":null,\"integrityImpact\":\"HIGH\",\"accessComplexity\":null,\"availabilityImpact\":\"HIGH\",\"confidentialityImpact\":\"HIGH\",\"scope\":\"UNCHANGED\",\"attackVector\":\"LOCAL\",\"baseSeverity\":\"HIGH\",\"userInteraction\":\"NONE\",\"attackComplexity\":\"LOW\",\"privilegesRequired\":\"LOW\"}",
    "PublishedDate": "2022-11-07T00:15:09.697",
    "ModifyDate": "2024-11-21T07:25:35.66",
    "VulnStatus": "Modified",
    "JsonData": "{\"id\": \"CVE-2022-42919\", \"summary\": \"Python 3.9.x before 3.9.16 and 3.10.x before 3.10.9 on Linux allows local privilege escalation in a non-default configuration. The Python multiprocessing library, when used with the forkserver start method on Linux, allows pickles to be deserialized from any user in the same machine local network namespace, which in many system configurations means any user on the same machine. Pickles can execute arbitrary code. Thus, this allows for local user privilege escalation to the user that any forkserver process is running as. Setting multiprocessing.util.abstract_sockets_supported to False is a workaround. The forkserver start method for multiprocessing is not the default start method. This issue is Linux specific because only Linux supports abstract namespace sockets. CPython before 3.9 does not make use of Linux abstract namespace sockets by default. Support for users manually specifying an abstract namespace socket was added as a bugfix in 3.7.8 and 3.8.3, but users would need to make specific uncommon API calls in order to do that in CPython before 3.9.\", \"created_at\": \"2022-11-07T00:15:09Z\", \"updated_at\": \"2024-11-21T07:25:35Z\", \"cvss\": {\"v2\": null, \"v3\": 7.8}, \"vendors\": {\"python\": [\"python\"], \"fedoraproject\": [\"fedora\"]}, \"cwes\": [\"NVD-CWE-noinfo\"], \"raw_nvd_data\": {\"id\": \"CVE-2022-42919\", \"metrics\": {\"cvssMetricV31\": [{\"type\": \"Primary\", \"source\": \"nvd@nist.gov\", \"cvssData\": {\"scope\": \"UNCHANGED\", \"version\": \"3.1\", \"baseScore\": 7.8, \"attackVector\": \"LOCAL\", \"baseSeverity\": \"HIGH\", \"vectorString\": \"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H\", \"integrityImpact\": \"HIGH\", \"userInteraction\": \"NONE\", \"attackComplexity\": \"LOW\", \"availabilityImpact\": \"HIGH\", \"privilegesRequired\": \"LOW\", \"confidentialityImpact\": \"HIGH\"}, \"impactScore\": 5.9, \"exploitabilityScore\": 1.8}]}, \"published\": \"2022-11-07T00:15:09.697\", \"references\": [{\"url\": \"https://github.com/python/cpython/compare/v3.10.8...v3.10.9\", \"tags\": [\"Release Notes\"], \"source\": \"cve@mitre.org\"}, {\"url\": \"https://github.com/python/cpython/compare/v3.9.15...v3.9.16\", \"tags\": [\"Release Notes\"], \"source\": \"cve@mitre.org\"}, {\"url\": \"https://github.com/python/cpython/issues/97514\", \"tags\": [\"Third Party Advisory\"], \"source\": \"cve@mitre.org\"}, {\"url\": \"https://github.com/python/cpython/issues/97514#issuecomment-1310277840\", \"tags\": [\"Third Party Advisory\"], \"source\": \"cve@mitre.org\"}, {\"url\": \"https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/FKGCQPIVHEAIJ77R3RSNSQWYBUDVWDKU/\", \"source\": \"cve@mitre.org\"}, {\"url\": \"https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/P2LHWWEI5OBQ6RELULMVU6KMDYG4WZXH/\", \"source\": \"cve@mitre.org\"}, {\"url\": \"https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/PI5DYIED6U26BGX5IRZWNCP6TY4M2ZGZ/\", \"source\": \"cve@mitre.org\"}, {\"url\": \"https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/QLUGZSEAO3MBWGKCUSMKQIRYJZKJCIOB/\", \"source\": \"cve@mitre.org\"}, {\"url\": \"https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/R6KGIRHSENZ4QAB234Z36HVIDTRJ3MFI/\", \"source\": \"cve@mitre.org\"}, {\"url\": \"https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/RDK3ZZBRYFO47ET3N4BNTKVXN47U6ICY/\", \"source\": \"cve@mitre.org\"}, {\"url\": \"https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/VCRKBB5Y5EWTJUNC7LK665WO64DDXSTN/\", \"source\": \"cve@mitre.org\"}, {\"url\": \"https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/XX6LLAXGZVZ327REY6MDZRMMP47LJ53P/\", \"source\": \"cve@mitre.org\"}, {\"url\": \"https://security.gentoo.org/glsa/202305-02\", \"source\": \"cve@mitre.org\"}, {\"url\": \"https://security.netapp.com/advisory/ntap-20221209-0006/\", \"tags\": [\"Third Party Advisory\"], \"source\": \"cve@mitre.org\"}, {\"url\": \"https://github.com/python/cpython/compare/v3.10.8...v3.10.9\", \"tags\": [\"Release Notes\"], \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}, {\"url\": \"https://github.com/python/cpython/compare/v3.9.15...v3.9.16\", \"tags\": [\"Release Notes\"], \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}, {\"url\": \"https://github.com/python/cpython/issues/97514\", \"tags\": [\"Third Party Advisory\"], \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}, {\"url\": \"https://github.com/python/cpython/issues/97514#issuecomment-1310277840\", \"tags\": [\"Third Party Advisory\"], \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}, {\"url\": \"https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/FKGCQPIVHEAIJ77R3RSNSQWYBUDVWDKU/\", \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}, {\"url\": \"https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/P2LHWWEI5OBQ6RELULMVU6KMDYG4WZXH/\", \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}, {\"url\": \"https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/PI5DYIED6U26BGX5IRZWNCP6TY4M2ZGZ/\", \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}, {\"url\": \"https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/QLUGZSEAO3MBWGKCUSMKQIRYJZKJCIOB/\", \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}, {\"url\": \"https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/R6KGIRHSENZ4QAB234Z36HVIDTRJ3MFI/\", \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}, {\"url\": \"https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/RDK3ZZBRYFO47ET3N4BNTKVXN47U6ICY/\", \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}, {\"url\": \"https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/VCRKBB5Y5EWTJUNC7LK665WO64DDXSTN/\", \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}, {\"url\": \"https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/XX6LLAXGZVZ327REY6MDZRMMP47LJ53P/\", \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}, {\"url\": \"https://security.gentoo.org/glsa/202305-02\", \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}, {\"url\": \"https://security.netapp.com/advisory/ntap-20221209-0006/\", \"tags\": [\"Third Party Advisory\"], \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}], \"vulnStatus\": \"Modified\", \"weaknesses\": [{\"type\": \"Primary\", \"source\": \"nvd@nist.gov\", \"description\": [{\"lang\": \"en\", \"value\": \"NVD-CWE-noinfo\"}]}], \"descriptions\": [{\"lang\": \"en\", \"value\": \"Python 3.9.x before 3.9.16 and 3.10.x before 3.10.9 on Linux allows local privilege escalation in a non-default configuration. The Python multiprocessing library, when used with the forkserver start method on Linux, allows pickles to be deserialized from any user in the same machine local network namespace, which in many system configurations means any user on the same machine. Pickles can execute arbitrary code. Thus, this allows for local user privilege escalation to the user that any forkserver process is running as. Setting multiprocessing.util.abstract_sockets_supported to False is a workaround. The forkserver start method for multiprocessing is not the default start method. This issue is Linux specific because only Linux supports abstract namespace sockets. CPython before 3.9 does not make use of Linux abstract namespace sockets by default. Support for users manually specifying an abstract namespace socket was added as a bugfix in 3.7.8 and 3.8.3, but users would need to make specific uncommon API calls in order to do that in CPython before 3.9.\"}, {\"lang\": \"es\", \"value\": \"Python 3.9.x anterior a la versi\\u00f3n 3.9.16 y 3.10.x anterior a la versi\\u00f3n 3.10.9 en Linux permite la escalada de privilegios locales en una configuraci\\u00f3n no predeterminada. La librer\\u00eda de multiprocesamiento Python, cuando se usa con el m\\u00e9todo de inicio forkserver en Linux, permite deserializar pickles de cualquier usuario en el mismo espacio de nombres de red local de la m\\u00e1quina, lo que en muchas configuraciones del sistema significa cualquier usuario en la misma m\\u00e1quina. Pickles puede ejecutar c\\u00f3digo arbitrario. Por lo tanto, esto permite escalar los privilegios del usuario local al usuario con el que se ejecuta cualquier proceso de forkserver. Como workaround se puede configurar multiprocessing.util.abstract_sockets_supported a False. El m\\u00e9todo de inicio de forkserver para multiprocesamiento no es el m\\u00e9todo de inicio predeterminado. Este problema es espec\\u00edfico de Linux porque solo Linux admite sockets de espacios de nombres abstractos. CPython anterior a 3.9 no utiliza sockets de espacio de nombres abstractos de Linux de forma predeterminada. Se a\\u00f1adi\\u00f3 soporte para usuarios que especifican manualmente un socket de espacio de nombres abstracto como correcci\\u00f3n de errores en 3.7.8 y 3.8.3, pero los usuarios tendr\\u00edan que realizar llamadas API poco comunes espec\\u00edficas para poder hacerlo en CPython antes de 3.9.\"}], \"lastModified\": \"2024-11-21T07:25:35.660\", \"configurations\": [{\"nodes\": [{\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:a:python:python:*:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"D90DBA72-F261-408A-B8DD-5D2DD7E5985F\", \"versionEndIncluding\": \"3.7.15\", \"versionStartIncluding\": \"3.7.3\"}, {\"criteria\": \"cpe:2.3:a:python:python:*:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"78F6D00F-54D9-4B4F-91CB-20F262F83570\", \"versionEndIncluding\": \"3.8.15\", \"versionStartIncluding\": \"3.8.3\"}, {\"criteria\": \"cpe:2.3:a:python:python:*:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"E28B140D-129A-4B9F-AC2B-F121E7EAD70C\", \"versionEndExcluding\": \"3.9.16\", \"versionStartIncluding\": \"3.9.0\"}, {\"criteria\": \"cpe:2.3:a:python:python:*:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"95E57263-9F7E-489E-9578-D89B584095B2\", \"versionEndExcluding\": \"3.10.9\", \"versionStartIncluding\": \"3.10.0\"}], \"operator\": \"OR\"}]}, {\"nodes\": [{\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:o:fedoraproject:fedora:35:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"80E516C0-98A4-4ADE-B69F-66A772E2BAAA\"}, {\"criteria\": \"cpe:2.3:o:fedoraproject:fedora:36:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"5C675112-476C-4D7C-BCB9-A2FB2D0BC9FD\"}, {\"criteria\": \"cpe:2.3:o:fedoraproject:fedora:37:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"E30D0E6F-4AE8-4284-8716-991DFA48CC5D\"}], \"operator\": \"OR\"}]}], \"sourceIdentifier\": \"cve@mitre.org\"}}\n",
    "LastUpdate": 0,
    "VendorL": [
      "python",
      "fedoraproject"
    ],
    "ProductL": [
      "python",
      "fedora"
    ],
    "Software_Repo_Mappings": []
  },
  "Software_Repo_Mapping_Conditions": [
    {
      "Software_Repo_Mapping_ConditionId": 0,
      "Software_Repo_MappingId": null,
      "Criteria": "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*",
      "VersionEndExcluding": null,
      "VersionEndIncluding": "3.7.15",
      "VersionStartExcluding": null,
      "VersionStartIncluding": "3.7.3",
      "Vulnerable": true,
      "Software_Repo_Mapping": null
    },
    {
      "Software_Repo_Mapping_ConditionId": 0,
      "Software_Repo_MappingId": null,
      "Criteria": "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*",
      "VersionEndExcluding": null,
      "VersionEndIncluding": "3.8.15",
      "VersionStartExcluding": null,
      "VersionStartIncluding": "3.8.3",
      "Vulnerable": true,
      "Software_Repo_Mapping": null
    },
    {
      "Software_Repo_Mapping_ConditionId": 0,
      "Software_Repo_MappingId": null,
      "Criteria": "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*",
      "VersionEndExcluding": "3.9.16",
      "VersionEndIncluding": null,
      "VersionStartExcluding": null,
      "VersionStartIncluding": "3.9.0",
      "Vulnerable": true,
      "Software_Repo_Mapping": null
    },
    {
      "Software_Repo_Mapping_ConditionId": 0,
      "Software_Repo_MappingId": null,
      "Criteria": "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*",
      "VersionEndExcluding": "3.10.9",
      "VersionEndIncluding": null,
      "VersionStartExcluding": null,
      "VersionStartIncluding": "3.10.0",
      "Vulnerable": true,
      "Software_Repo_Mapping": null
    }
  ]
}